Point-of-No-Return/Character.h

```diff
@@ -3,48 +3,148 @@
 
 #include "Object.h"
 
-//キャラクタークラス
-class Character :public Object
+/**
+ * @brief 方向
+ */
+enum class Direction
+{
+	//! 上
+	Up,
+	//! 下
+	Down,
+	//! 右
+	Right,
+	//! 左
+	Left
+};
+
+/**
+ * @brief キャラクタークラス
+ */
+class Character : public Object
 {
 public:
+	/**
+	 * @brief コンストラクタ
+	 * @param name キャラクターの名前
+	 */
 	explicit Character(const std::string& name):name(name) {}
+
+	/**
+	 * @brief	座標を修正する
+	 * @param	direction　　　座標修正する方向
+	 * @param	blockPosition　当たっているブロックの座標
+	 * @details	HitCheckの中で呼び出す
+	 */
+	virtual void CorrectCoordinate(Direction direction, const Vec2& blockPosition) {};
+protected:
+
+	//! 前の座標を保存する
+	Vec2 previous = { 0,0 };
+
+	// TODO: あとでprimitive型wrapします
+	float gravity = 0;
+
 private:
+
+	//! キャラクターの名前
 	Name name = { "None" };
 };
 
-//主人公クラス
-class Hero :public Character
+/**
+ * @brief 主人公クラス
+ * @details コピーコンストラクタと演算子オーバーロードの隠蔽
+ */
+class Hero : public Character
 {
 public:
+	// void HitBlock(std::vector<Vec2> vec2);
+
+	/**
+	 * @brief コンストラクタ
+	 * @param name 主人公の名前
+	 */
 	Hero(const std::string& name):Character(name)
 	{
+		
 		Initialize();
 	}
-
+	
+	/**
+	 * @brief  主人公の情報を更新する
+	 */
 	void Update();
 
+	/**
+	 * @brief	座標を修正する
+	 * @param	direction　　　座標修正する方向
+	 * @param	blockPosition　当たっているブロックの座標
+	 * @details	HitCheckの中で呼び出す
+	 */
+	void CorrectCoordinate(Direction direction, const Vec2& blockPosition) override;
```

Comment:
「基底クラスでvirtualならシグネチャーが同一の関数は暗黙的にvirtual」となる仕様を利用した書き方ですね。

読み手にもよりますが、
```suggestion
	virtual void CorrectCoordinate(Direction direction, const Vec2& blockPosition) override;
```
と明示的に書いてあげる方が良いかも知れません。
https://qiita.com/yumetodo/items/f3ceb1a64a1540f85568

Reply:
修正しました。https://github.com/human-osaka-game-2019/Point-of-No-Return/pull/39/commits/2f49a544f12267523e493ada7f5401f757254692